backend/portal/serializers.py

```diff
@@ -223,106 +212,25 @@ def get_image_url(self, obj):
         else:
             return image.url
 
-    class Meta:
+    class Meta(ContentSerializer.Meta):
         model = Post
         fields = (
-            "id",
-            "club_code",
+            *ContentSerializer.Meta.fields,
             "title",
             "subtitle",
             "post_url",
             "image",
             "image_url",
-            "created_date",
-            "start_date",
-            "expire_date",
-            "club_comment",
-            "admin_comment",
-            "status",
-            "target_populations",
         )
-        read_only_fields = ("id", "created_date", "target_populations")
```

Comment:
The reason you put target_population as read_only for Posts is because otherwise, the built in serializer tries to validate / set it, but it cannot succesfully validate the data coming from a form causing errors. (hence you marked it as read_only and then manually did it yourself after the fact).

I've tested post with image + non-empty and empty list together... after dealing with the edge case in is_valid in the Post serializer, it should be treating the target_population as any other field now

Reply:
Bet thanks